dbms/src/Formats/ParsedTemplateFormatString.cpp

```diff
@@ -16,20 +16,10 @@ namespace ErrorCodes
 
 ParsedTemplateFormatString::ParsedTemplateFormatString(const FormatSchemaInfo & schema, const ColumnIdxGetter & idx_by_name)
 {
-    try
-    {
-        ReadBufferFromFile schema_file(schema.absoluteSchemaPath(), 4096);
-        String format_string;
-        readStringUntilEOF(format_string, schema_file);
-        parse(format_string, idx_by_name);
-    }
-    catch (DB::Exception & e)
-    {
-        if (e.code() != ErrorCodes::INVALID_TEMPLATE_FORMAT)
```

Comment:
Not sure what's the point, if the error was not in the wrong template format, reporting how exactly we failed to parse the format doesn't help. In a simple case of nonexistent file, it was reporting that it could not open the file, followed by the description of empty format.

Reply:
Even if format string syntax is correct, `parse(...)` may throw `NOT_FOUND_COLUMN_IN_BLOCK` or `CANNOT_PARSE_QUOTED_STRING` or something else for wrong column or substitution name without any diagnostic info. This try-catch is needed to add more information in such cases. But operations with file and buffer should not be in `try` block, right.